aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -1588,6 +1674,44 @@ std::optional<c10::ScalarType> out_dtype) {
 #endif
 }
 
+Tensor _bmm_dtype_cuda(const Tensor& batch1, const Tensor& batch2, const at::ScalarType out_dtype) {
+  Scalar beta(0.0);
+  Scalar alpha(1.0);
+  IntArrayRef batch1_sizes = batch1.sizes();
+  IntArrayRef batch2_sizes = batch2.sizes();
+
+  Tensor out = at::empty({batch1_sizes[0], batch1_sizes[1], batch2_sizes[2]}, batch1.options().dtype(out_dtype));
```

Comment:
check that input and out dtypes are matching, we support only limited amount for cross-dtype matmuls (e.g. no bf16bf16->fp64 etc)